homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -439,13 +423,8 @@ def white_value_received(msg):
             self._white_value = percent_white * 255
             self.async_write_ha_state()
 
-        if self._topic[CONF_WHITE_VALUE_STATE_TOPIC] is not None:
-            topics[CONF_WHITE_VALUE_STATE_TOPIC] = {
-                "topic": self._topic[CONF_WHITE_VALUE_STATE_TOPIC],
-                "msg_callback": white_value_received,
-                "qos": self._config[CONF_QOS],
-            }
-        elif (
+        add_topic(CONF_WHITE_VALUE_STATE_TOPIC, white_value_received)
+        if (
```

Comment:
The `elif` was not correct, we should restore the state if `self._optimistic_white_value` is `True`